surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -48,15 +54,28 @@ private boolean isUsageSet() {
         for (final String strategy : strategies) {
 
             final TestExecutionPlanner plannerForStrategy = testExecutionPlannerLoader.getPlannerForStrategy(strategy);
-            final List<? extends Class<?>> tests = plannerForStrategy.getTests().stream().map(testClass -> {
-                try {
-                    return Class.forName(testClass);
-                } catch (ClassNotFoundException e) {
-                    throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
-                }
-            }).collect(Collectors.toList());
+            final List<? extends Class<?>> tests = plannerForStrategy.getTests()
+                .stream().filter(this::validateClassPathAndTestClass)
+                .map(testClass -> {
+                    try {
+                        return Class.forName(testClass);
+                    } catch (ClassNotFoundException e) {
+                        throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
+                    }
+                }).collect(Collectors.toList());
             orderedTests.addAll(tests);
         }
         return orderedTests;
     }
+
+    private boolean validateClassPathAndTestClass(String testClass) {
+        Class<?> aClass;
+        try {
+            aClass = testClassLoader.loadClass(testClass);
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+
+        return jUnit4TestChecker.accept(aClass);
```

Comment:
I checked in the bare project and by default surefire will ignore `Test*` from src/main. For us it is failing, but I believe the fix is not to check if a class is eg. JUnit4 test, because this should be done by the test runner we delegate to. So this means we are having an issue with the way how we instrument test runner that the check is not invoked.